pkg/types/models.go

```diff
@@ -280,17 +309,17 @@ func (h *Head) UnmarshalJSON(bs []byte) error {
 		return nil
 	}
 
-	h.Hash = jsonHead.Hash
+	h.Hash = common.BytesToHash(jsonHead.Hash.Bytes())
```

Comment:
We can just cast it to common.Hash. This might help us to avoid allocation during conversion of [32]bytes to []bytes

Reply:
Think common.Hash is a type so we can't easily cast to it, my go experiences isn't the best but think the only way to do that and avoid allocation would be something like this which i'd rather avoid. 

If theres a better way happy to implement it

```
func (h Hash) ToCommonHash() common.Hash {
	return *(*common.Hash)(unsafe.Pointer(&h))
}
```

Reply:
common.Hash is a type alias to the `[HashLength]byte`, so casting should work. Just tried it locally - works for me
```suggestion
	h.Hash = common.Hash(jsonHead.Hash)
```